dask/array/tests/test_random.py

```diff
@@ -161,6 +161,18 @@ def test_random_all():
     da.random.standard_normal(size=5, chunks=3).compute()
     da.random.standard_t(2, size=5, chunks=3).compute()
 
+@pytest.mark.skipif(not hasattr(np,'broadcast_to'),
+                    reason='requires numpy 1.10 method "broadcast_to"' )
+def test_array_broadcasting():
+    arr = np.arange(6).reshape((2,3))
+    assert da.random.poisson(arr, chunks=3).compute().shape == (2,3)
+    assert da.random.normal(arr, 2, chunks=3).compute().shape == (2,3)
+    assert da.random.normal(np.ones((1,4)), np.ones((3,1)),
+                            chunks=3).compute().shape == (3,4)
+    assert da.random.normal(np.ones((1,4)), np.ones((3,1)), size=(2, 3,4),
+                            chunks=3).compute().shape == (2,3,4)
+    with pytest.raises(ValueError):
+        da.random.normal(arr, np.ones((3,1)), size=(2,3,4), chunks=3)
```

Comment:
We should probably have tests that use dask.arrays in here as well.  Generally speaking if there isn't a test stressing the functionality it's very easy for someone else to come along and break it later without anyone knowing.  I serve as general watchdog for this kind of breaking change but I'm not involved/attentive enough in all of the code to catch everything.